### PR TITLE
Disentangle amici and roadrunner

### PR DESCRIPTION
**Pull request by PaulJonasJost**
When installing only pypesto[roadrunner], there are additional amici dependencies, mainly occurring in hierarchical (basically all of them typings, thus fixed through string type annotation)

### CODE REVIEW

**Review by dweindl — COMMENTED** [review]
Thanks. For all touched Python modules, I'd prefer adding `from __future__ import annotations` instead of the changes made here. This is will cover any future changes in type annotations there.

File: setup.cfg [hunk 1]
```diff
@@ -97,6 +97,8 @@ amici =
     amici >= 0.21.0
 roadrunner =
     libroadrunner >= 2.7.0
+    petab >= 0.4.0
```
Comment (dweindl) [inline setup.cfg:100]:
```suggestion
    %(petab)s
```

**Review by dweindl — APPROVED** [review]
:+1:

File: pypesto/hierarchical/semiquantitative/solver.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (FFroehlich) [inline pypesto/hierarchical/semiquantitative/solver.py:1]: unused?

**Review by FFroehlich — APPROVED** [review]

File: pypesto/hierarchical/semiquantitative/solver.py (hunk 2 above)
Comment (PaulJonasJost) [reply pypesto/hierarchical/semiquantitative/solver.py:1]: needed for the ParameterMapping

**Review by m-philipps — APPROVED** [review]
Looks good as far as I can tell.

**Pull request merged**